library/src/main/java/com/google/android/exoplayer/MediaCodecUtil.java

```diff
@@ -55,6 +57,14 @@ private DecoderQueryException(Throwable cause) {
 
   private static final HashMap<CodecKey, Pair<String, CodecCapabilities>> codecs = new HashMap<>();
 
+  private static final HashMap<String, ArrayList<String>> mimeTypeAlternatives = new HashMap<>();
+
+  static {
+    // It's possible that some devices use alternative MimeTypes for codecs to those used internally. We
+    // therefore should consider them when attempting to locate the associated MediaCodec.
+    mimeTypeAlternatives.put(MimeTypes.AUDIO_DTS_HD, new ArrayList<>(Arrays.asList(new String[]{"audio/dtshd"})));
```

Comment:
It's likely that other alternatives will be added but I wanted to test on these devices before suggesting the changes. This specific alternative came from the Minix Neo device. Interestingly, it only provides a DTS-HD MediaCodec but not a DTS one (strange).